File: R/8kyu/8kyu-basic-mathematical-operations.r
# Basic Mathematical Operations
# Language: R
# Difficulty: 8 kyu

# Solution
basic_op <- function(operator, value1, value2){
  eval(parse(text = paste(as.character(value1), operator, as.character(value2))))
}

# Refactored soltuin
basic_op <- function(operator, value1, value2){
  eval(parse(text = paste(value1, operator, value2)))
}


# Tests
test_that('Basic tests', {
  expect_equal(basic_op('+', 4, 7), 11)
  expect_equal(basic_op('-', 15, 18), -3)
  expect_equal(basic_op('*', 5, 5), 25)
  expect_equal(basic_op('/', 49, 7), 7)
})

File: Python/8kyu/8kyu-basic-mathematical-operations.py
# Basic Mathematical Operations
# Language: Python
# Difficulty: 8 kyu

def basic_op(operator, value1, value2):
    return eval(str(value1) + operator + str(value2))


# Tests
import codewars_test as test
from solution import basic_op

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(basic_op('+', 4, 7), 11)
        test.assert_equals(basic_op('-', 15, 18), -3)
        test.assert_equals(basic_op('*', 5, 5), 25)
        test.assert_equals(basic_op('/', 49, 7), 7)